src/hooks.c

```diff
@@ -0,0 +1,91 @@
+/*
+* Copyright (C) the libgit2 contributors. All rights reserved.
+*
+* This file is part of libgit2, distributed under the GNU GPL v2 with
+* a Linking Exception. For full terms see the included COPYING file.
+*/
+
+#include "common.h"
+#include "path.h"
+#include "hooks.h"
+#include "repository.h"
+
+static git_hook_callback registered_callbacks[GIT_HOOK_TYPE_MAXIMUM_SUPPORTED];
+
+int git_repository_hook_get(git_repository_hook **hook_out, git_repository *repo, git_hook_type type)
+{
+    git_repository_hook *hook = NULL;
+    const char *file_name;
+
+    assert(hook_out);
+    assert(repo);
+    assert(type >= 0 && type <= GIT_HOOK_TYPE_MAXIMUM_SUPPORTED);
+
+    hook = git__calloc(1, sizeof(git_repository_hook));
+    GITERR_CHECK_ALLOC(hook);
+
+    hook->type = type;
+
+    git_buf_init(&hook->path, 0);
+    if (git_buf_joinpath(&hook->path, repo->path_repository, GIT_HOOKS_DIRECTORY_NAME) != 0) {
+        git_repository_hook_free(hook);
+        return GIT_ERROR;
+    }
+
+    file_name = supported_hooks[type];
+    if (git_path_contains_file(&hook->path, file_name)) {
+        hook->exists = GIT_HOOK_TRUE;
```

Comment:
The file doesn't just have to exist, it also needs to be executable by the current user. This may not be an issue on Windows but on the other OSs, git won't execute a hook unless the file itself is executable as it's the OS' job to figure out how to execute the hook itself.